FeedStoreChallenge/CoreDataFeedStore.swift

```
@@ -0,0 +1,198 @@
+			let fetchRequest = NSFetchRequest<NSFetchRequestResult>(entityName: "CoreDataFeedImage")
+			let batchDeleteRequest = NSBatchDeleteRequest(fetchRequest: fetchRequest)
+
+			do {
+				try selfForBlcok.context.execute(batchDeleteRequest)
+				completion(nil)
+			} catch  {
+				completion(error)
+			}
+		}
+	}
+	
+	func insert(_ feed: [LocalFeedImage], timestamp: Date, completion: @escaping InsertionCompletion) {
+		let selfForBlcok = self
+		self.context.perform {
+			selfForBlcok.deleteCacheItems()
+			selfForBlcok.storeCache(imgs: feed.map{ CodableFeedImage(image: $0) }, date: timestamp, completion: completion)
+		}
+	}
+	
+	func retrieve(completion: @escaping RetrievalCompletion) {
+		self.context.perform {
+			if let result = self.fetchFeedImage() {
+				completion(.found(feed: result.localFeed, timestamp: result.timestamp))
+			}
+			else {
+				completion(.empty)
+			}
+		}
+	}
+	
+	
+	
+	private func storeCache( imgs : [CodableFeedImage],date: Date, completion: @escaping InsertionCompletion) {
+		for img in imgs {
+			let feed =  CoreDataFeedImage(context: context)
+			feed.date = date
+			feed.id = img.id
+			feed.imageInfo = img.description
+			feed.location = img.location
+			feed.url = img.url
+			
+			do {
+				try context.save()
+			} catch let error as NSError {
+				completion(error)
+				print("error \(error), \(error.userInfo)")
+			}
+		}
+		completion(nil)
+	}
+	
+	private func fetchFeedImage() -> Cache? {
+		var feeds = [CodableFeedImage]()
+		do {
+			let fetchRequest = 	NSFetchRequest<NSFetchRequestResult>(entityName: "CoreDataFeedImage")
+
+			guard let result = try context.fetch(fetchRequest) as? [CoreDataFeedImage], result.count > 0 else { return nil }
+			
+			feeds = result.map {
+				CodableFeedImage(image: LocalFeedImage(id: $0.id, description: $0.imageInfo!, location: $0.location!, url: $0.url!))
+			}
+			let cache = Cache.init(feed: feeds, timestamp: (result.first?.date)!)
+			return cache
+
+		} catch {
+			return nil
+		}
+	}
```

This method is ignoring errors. So if there's an error, the clients won't know about it. This leads to undefined behavior at runtime.

If there's an error, this method should throw it so clients can handle the error. 

Moreover, there are many force unwrap with `!`. This will lead to runtime errors (crashes).